libs/vkd3d/swapchain.c

```diff
@@ -2108,6 +2233,192 @@ static CONST_VTBL struct IDXGIVkSwapChainFactoryVtbl dxgi_vk_swap_chain_factory_
     dxgi_vk_swap_chain_factory_CreateSwapChain,
 };
 
+bool dxgi_vk_swap_chain_low_latency_enabled(struct dxgi_vk_swap_chain* chain)
+{
+    return chain->present.low_latency_state.mode;
+}
+
+HRESULT dxgi_vk_swap_chain_latency_sleep(struct dxgi_vk_swap_chain* chain)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &chain->queue->device->vk_procs;
+
+    VkLatencySleepInfoNV latency_sleep_info;
+    VkSemaphoreWaitInfo sem_wait_info;
+
+    if (chain->present.low_latency_state.mode)
+    {
+        // Increment the low latency sem value before the wait
+        chain->present.low_latency_sem_value++;
+
+        memset(&latency_sleep_info, 0, sizeof(latency_sleep_info));
+        latency_sleep_info.sType = VK_STRUCTURE_TYPE_LATENCY_SLEEP_INFO_NV;
+        latency_sleep_info.pNext = NULL;
+        latency_sleep_info.signalSemaphore = chain->present.low_latency_sem;
+        latency_sleep_info.value = chain->present.low_latency_sem_value;
+
+        pthread_mutex_lock(&chain->present.low_latency_lock);
```

Comment:
What is the purpose of this lock?

Reply:
Sorry that the purpose wasn't clear. I should have had a comment near the locks definition that explains its purpose. I will add that when addressing the rest of the review feedback. The main purpose of this lock is to prevent using the swapchain handle during swapchain recreation. I will update the code in dxgi_vk_swap_chain_present_callback to only take the lock when the swapchain needs to be recreated.